src/raw_node.rs

```diff
@@ -141,8 +138,8 @@ impl Ready {
             }
             rd.hs = Some(hs);
         }
-        if raft.raft_log.unstable.snapshot.is_some() {
-            rd.snapshot = raft.raft_log.unstable.snapshot.clone().unwrap();
+        if let Some(snapshot) = &raft.raft_log.unstable.snapshot {
+            rd.snapshot = snapshot.clone();
```

Comment:
It seems that if the snapshot is `Some<[]>` then it will still be cloned where it doesn't need to be.